### PR TITLE
[RJA-1067][feature]: UI for variables grouping.

### PR DESCRIPTION
**Pull request by ddurnev**
Required groups not supported yet.

### CODE REVIEW

File: ui/src/main/resources/app/modules/createenv.js [hunk 1]
```diff
@@ -212,9 +212,14 @@ function(genesis, backend,  status, variables, gtemplates, validation, Backbone,
 
   var EnvironmentParametersStep = Step.extend({
     template: "app/templates/createenv/environment_settings.html",
+    varTemplate: "app/templates/common/variables.html",
     errorTemplate: "app/templates/createenv/environment_settings_error.html",
     preconditionErrorTemplate: "app/templates/createenv/preconditions_error.html",
 
+    events: {
+      "click .group-radio": function(e) {variables.groupVarSelected(e, this, this.variables);}
```
Comment (bugzmanov) [inline ui/src/main/resources/app/modules/createenv.js]: bad styling.

  on-line function declaration suites for those places where in scala you'd use closures with one statement in it, like map, foreach.
callbacks declaration is not the case.

File: ui/src/main/resources/app/modules/createenv.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ function(genesis, backend,  status, variables, gtemplates, validation, Backbone,
     },
 
     createEnvironment: function() {
-      if($('#environment-settings').valid()) {
+      if($('#workflow-parameters-form').valid()) {
```
Comment (bugzmanov) [inline ui/src/main/resources/app/modules/createenv.js]: is global $ really neeaded here?

File: ui/src/main/resources/app/modules/env_details/env_details.js [hunk 3]
```diff
@@ -380,6 +380,11 @@ function (genesis, backend, poller, status, EnvHistory, EnvAccess, variables, gt
 
   var ExecuteWorkflowDialog = Backbone.View.extend({
     template: "app/templates/environment_variables.html",
+    templateVars: "app/templates/common/variables.html",
+
+    events: {
+      "click .group-radio": function(e) {variables.groupVarSelected(e, this, this.workflow.variables);}
```
Comment (bugzmanov) [inline ui/src/main/resources/app/modules/env_details/env_details.js]: bad style

File: ui/src/main/resources/app/variables.js [hunk 4]
```diff
@@ -143,6 +143,18 @@ function(genesis, status, $, _) {
             });
         });
       });
+    },
+
+    groupVarSelected: function(event, view, variables) {
+       var $currentTarget = view.$(event.currentTarget);
+       var group = $currentTarget.attr('name');
+       var name = $currentTarget.attr('data-var-name');
+       view.$("[id=" + name + "]").show();
```
Comment (bugzmanov) [inline ui/src/main/resources/app/variables.js]: why not  view.$("#" + name)  ?

File: ui/src/main/resources/app/variables.js [hunk 5]
```diff
@@ -143,6 +143,18 @@ function(genesis, status, $, _) {
...
+       _.each(variables.filter(function(v){return group == v.group;}), function (x) {
+         if (x.name !== name) {
```
Comment (bugzmanov) [inline ui/src/main/resources/app/variables.js]: why x.name !== name is not in filter expression?

**Comment by ddurnev** [comment]
Updated.

**Comment by ddurnev** [comment]
Updated last commit.

**Comment by ddurnev** [comment]
updated

**Comment by bugzmanov** [comment]
Margins are still problem. For checkboxes and radio buttons - left margin in this particular form should be 0 I suppose (or regular inputs and comboboxes should have the same)

in group variables I suppose inputs should not have top margin.

**Comment by ddurnev** [comment]
Updated.

**Pull request merged**